### ISSUE TITLE
@types/sequelize do not follow addIndex types from the documentation

### ISSUE THREAD

**Issue by oktapodia** [DefinitelyTyped/DefinitelyTyped#19182]
Hello

the `addIndex` method do not follow the sequelize v4 documentation ([link here](http://docs.sequelizejs.com/manual/tutorial/migrations.html#addindex-tablename-attributes-options-))
  - Authors: @samuelneff @CodeAnimal @drinchev @babolivier @kukoo1

**Comment by oktapodia** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/19181

**Comment by bkw** [comment]
I did a bit of research on this that I wanted to share here in case somebody wants to dive down this rabbit hole (I ultimately decided against doing so):

The [current sequelize docs](http://docs.sequelizejs.com/class/lib/query-interface.js~QueryInterface.html#instance-method-addIndex) lists the following options for `queryInterface.addIndex()`:

* fields
* unique
* using
* type
* name
* where

Current @types/sequelize exposes the following list instead (`DefineIndexOptions`):

* indicesType
* indexName
* parser
* indexType
* logging
* where

There's not even much overlap.  The documented ones seem to be closer to `DefineIndexesOptions`  (note the plural) than `DefineIndexOptions`. The current form looks like there is no way to specify the fieldnames, since it has no option `fields`.

Also, the [current sequelize docs](http://docs.sequelizejs.com/class/lib/query-interface.js~QueryInterface.html#instance-method-addIndex) make no mention of a second argument `attributes`, although checking [the  source code](https://github.com/sequelize/sequelize/blob/11d4a2091ebe28a0f6c364e022552835665bf965/lib/query-interface.js#L589-L607) reveals that `attributes` does actually exist and can be used to specify the fields for the index. It should probably be renamed to `fields` in that context.

Looking at the sequelize source confirms that the sequelize docs are - although incomplete - closer to the truth than what our definitions currently imply.

I briefly glanced over [addIndexQuery](https://github.com/sequelize/sequelize/blob/bb874f5e7e0cc65a3ef70edd60d8ef24685fe3ec/lib/dialects/abstract/query-generator.js#L519-L618) and could not find any use of `indicesType`, `indexName` or `indexType`. Instead it looks at `concurrently`, `prefix` and `operator`, neither of which are present in any of the two interfaces.

The options are [later](https://github.com/sequelize/sequelize/blob/11d4a2091ebe28a0f6c364e022552835665bf965/lib/query-interface.js#L606) being passed to sequelize.query(), which adds a whole new set of possibilities, like `logging`, which - as it happens - is being tested for in the tests for sequelize in this repo. When I found this, I decided to stop.

Maybe this helps the next poor soul who wants to tackle this. I'm glad I learned that the optional `attributes` argument is actually the field names for the index, and that allows my use case (adding a simple index in a migration using QueryInterface) to work, although the types for the options are way off.

### TL;DR:
The types for options are currently not correct.
If you are looking for a way to add an index, use ```queryInterface.addIndex(tableName, [field1, field2])```

**Comment by orta** [comment]
Hi thread, we're moving DefinitelyTyped to use [GitHub Discussions](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377) for conversations the `@types` modules in DefinitelyTyped.

To help with the transition, we're closing all issues which haven't had activity in the last 6 months, which includes this issue. If you think closing this issue is a mistake, please pop into the [TypeScript Community Discord](https://discord.gg/typescript) and mention the issue in the `definitely-typed` channel.
